From 393c22566b04d28adc28c0ca789a8c0f8eedbd69 Mon Sep 17 00:00:00 2001
From: jeremyd2019 <github@jdrake.com>
Date: Wed, 7 May 2025 12:28:11 -0700
Subject: [PATCH] [LLD][CMake] fix testing standalone build without installed
 llvm-lit. (#138575)

This block was copied from clang/CMakeLists.txt.
---
 lld/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lld/CMakeLists.txt b/lld/CMakeLists.txt
index 64c9f23805..a8b677a703 100644
--- a/lld/CMakeLists.txt
+++ b/lld/CMakeLists.txt
@@ -63,6 +63,9 @@ if(LLD_BUILT_STANDALONE)
     if(EXISTS ${LLVM_MAIN_SRC_DIR}/utils/lit/lit.py)
       # Note: path not really used, except for checking if lit was found
       set(LLVM_EXTERNAL_LIT ${LLVM_MAIN_SRC_DIR}/utils/lit/lit.py)
+      if(EXISTS ${LLVM_MAIN_SRC_DIR}/utils/llvm-lit)
+        add_subdirectory(${LLVM_MAIN_SRC_DIR}/utils/llvm-lit utils/llvm-lit)
+      endif()
       if(NOT LLVM_UTILS_PROVIDED)
         add_subdirectory(${LLVM_MAIN_SRC_DIR}/utils/FileCheck utils/FileCheck)
         add_subdirectory(${LLVM_MAIN_SRC_DIR}/utils/not utils/not)
-- 
2.50.1.windows.1

